src/server/frontend_xwayland/xwayland_wm_surface.cpp

```diff
@@ -125,9 +142,9 @@ void mf::XWaylandWMSurface::net_wm_state_client_message(uint32_t const (&data)[5
 
         new_window_state = window_state;
 
-        for (size_t i = 0; i < length_of(message->properties); i++)
+        for (xcb_atom_t const property : {prop_a, prop_b})
         {
-            if (xcb_atom_t const property = message->properties[i]) // if there is only one property, the 2nd is 0
+            if (property) // if there is only one property, the 2nd is 0
```

Comment:
I fail to see how the properties array (red) is cleaner than initializer list (green)